### DISCUSSION TITLE
Why does axmol needs libGLESv2.dll libEGL.dll on windows desktop? isn't it using OpenGL directly ?

### DISCUSSION THREAD

**Discussion by meiry** [axmolengine/axmol#838] [Q&A]
Why there are libEGL.dll libGLESv2.dll d3dcompiler_47.dll 
on windows dyployment ? 
Isn't the engine using OpenGL directly ?  what is the mimum OpenGL version the engine using ?

**Comment by halx99** [comment]
The  `libGLESv2.dll` and `libEGL.dll` is built from https://github.com/google/angle, not mandatory, only when you specific `-DAX_USE_COMPAT_GL=TRUE`  to generate native project with cmake, refer to https://github.com/axys1/axys/blob/dev/CMakeOptions.md

**Reply by meiry** [reply]
Thanks 
building project like this : cmake -B build_x64 -A x64 "-DAX_PREBUILT_DIR=build_x64"
still bring the dll's 
![image](https://user-images.githubusercontent.com/1107316/188068976-e4bb17f1-8a91-4bab-bbbb-8656f44b4298.png)

**Reply by halx99** [reply]
![image](https://user-images.githubusercontent.com/6977319/188070957-13b02cfd-c433-4b98-be20-141ff3056343.png)

**Reply by meiry** [reply]
Tnx , ok i did one more expirement :
run this :
```
axys new -d c:\dev\my\cpp\axys_projects -l cpp  isoTest2
```
this time i run this cmake command : 
```
cmake -B build_x64 -A x64 "-DAX_PREBUILT_DIR=build_x64 -DAX_ENABLE_MSEDGE_WEBVIEW2=FALSE -DAX_USE_COMPAT_GL=FALSE -DAX_ENABLE_EXT_SPINE=FALSE -DAX_ENABLE_EXT_COCOSTUDIO=FALSE"
```
result :  what do i miss ?
![image](https://user-images.githubusercontent.com/1107316/188071989-35fe8408-b0da-436f-89ea-6fff5f4a87e1.png)

**Comment by halx99** [comment]
Do you use latest commit?

**Reply by meiry** [reply]
@halx99 
yes :
![image](https://user-images.githubusercontent.com/1107316/188114408-4ab72991-9475-452a-8afb-bb1b9ded3e55.png)

when i debug the initiation of the project i can see it is using the latest installed in my pc 
![image](https://user-images.githubusercontent.com/1107316/188114666-99d547f8-bf07-46a2-9ff9-431f17a9a74c.png)

but when i delete the dlls it all works fine ... so i guess the cmake copy the unnecessary dll 
![image](https://user-images.githubusercontent.com/1107316/188115407-5bee62eb-7f1f-4cb3-a010-964b54206df0.png)

**Comment by halx99** [comment]
Seems your command line iincorrect

**Comment by halx99** [comment]
Yes, shoud fixed by https://github.com/axys1/axys/commit/a61a54c6be1531d31111bd4696abcab95b42e245

**Comment by meiry** [comment]
@halx99  tnx , 
What about : -DAX_ENABLE_MSEDGE_WEBVIEW2=FALSE
i dont expect to see WebView2Loader.dll while i set it to not use the webview

**Comment by halx99** [comment]
The latest commit will check AX_ENABLE_MSEDGE_WEBVIEW2 in all cmake files, but you needs manually delete `WebView2Loader.dll` if your previous build without `-DAX_ENABLE_MSEDGE_WEBVIEW2=FALSE`
